src/Command/HttpClientBatch.php

```diff
@@ -34,29 +34,29 @@ protected function configure()
 
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        $delays = [1, 3, 3, 5, 10];
+        // this command runs requests concurrently and read them in *request* order in 10s too
+
+        $delays = [10, 3, 3, 5, 1];
         $responses = [];
 
         foreach ($delays as $offset => $delay) {
+            $url = sprintf('https://reqres.in/api/users/%d?delay=%d', $offset + 1, $delay);
+            $responses[] = $this->httpClient->request('GET', $url);
+        }
+
+        /** @var ResponseInterface $response */
+        foreach ($responses as $url => $response) {
             try {
-                $responses[] = $this->httpClient->request(
-                    'GET',
-                    sprintf('https://reqres.in/api/users/%d?delay=%d', $offset + 1, $delay)
-                );
+                $data = $response->toArray();
```

Comment:
responses throw, not the request method (unless one is using a non-lazy implementation, but hey don't use any of them if one exists :) )